crates/pera-core/src/signature_mpc/mpc_manager.rs

```diff
@@ -19,6 +30,12 @@ struct MPCInstance {
     /// The channel to send message to this instance
     input_receiver: Option<mpsc::Sender<MPCInput>>,
     pending_messages: Vec<MPCInput>,
+    language_public_parameters:
+        maurer::language::PublicParameters<{ maurer::SOUND_PROOFS_REPETITIONS }, Lang>,
```

Comment:
It does consume redundant memory, but to resolve this we need to change the proof's repo `new_proof_round_party` function to receive an Arc instead of consuming the public parameters. @ycscaly Do you want me to change this?

Reply:
I removed it, there was no need for it apperantly